app/success/page.tsx

```diff
@@ -0,0 +1,46 @@
+import Link from 'next/link';
+import { redirect } from 'next/navigation';
+import { stripeClient } from '../../utils/client';
+
+type Props = {
+  searchParams: { sessionId: string };
+};
+
+export const metadata = { title: 'Success', description: 'Success Page' };
+
+export default async function SuccessPage({ searchParams }: Props) {
+  let session;
+  try {
+    session = await stripeClient.checkout.sessions.retrieve(
+      searchParams.sessionId,
+    );
+  } catch {
+    return redirect('/');
```

Comment:
I don't think you need to return here, `redirect` throws internally to stop execution